### ISSUE TITLE
HMDB XML data loading

### ISSUE THREAD

**Issue by gmhhope** [jorainer/SpectraTutorials#3]
Hi Johannes,

You see I am your fans now. I am testing your docker container for doing MS/MS search. However, I encountered issues with loading the hmdb spectra XML data. 

In brief, I have downloaded the `All Spectra Files (XML)` from https://hmdb.ca/downloads. And then I put into a local folder and then I run your docker container by: 
```
docker run -e PASSWORD=bioc -p 8787:8787 -v /Users/gongm/Documents/projects/test/SpectraTutorials/SpectraTutorials/data/:/home/rstudio/external jorainer/spectra_tutorials:latest
```
So that I can also access the spectra data in `../external/hmdb_all_spectra/`. However, when I run the below r script, the `fls` did not load correctly. And thus it gives me `character(0)`.

```{r hmdb-import, eval = FALSE}
library(MsBackendHmdb)

#' Get all MS2 spectra xml files and import data
fls <- dir("../external/hmdb_all_spectra/", full.names = TRUE, pattern = "ms_ms")
hmdb <- Spectra(fls, source = MsBackendHmdbXml(), nonStop = TRUE)
```
```
fls
character(0)
```

Do you know what is going wrong? And the loading did take at least 6 minutes before it ends.

Thanks very much always for your continuos help!

Thanks,
Minghao Gong

**Comment by gmhhope** [comment]
Later today, I did notice you have precompiled hmdb from your package:
```{r hmdb-load, echo = FALSE}
#' Load the pre-generated Spectra object with spectra from HMDB.
library(MsBackendHmdb)
data("hmdb", package = "SpectraTutorials")
```
However, it will be great if you could still guide me through the previous question, as the database will continue to update. 

But again I love to see your youtube video: https://www.youtube.com/watch?v=W6JISCrAtk0! It was fascinating for putting all these. Thanks very much!

Best,
Minghao

**Comment by jorainer** [comment]
Thanks Minghao!

hm, this is strange. I guess this has to do with the paths not being correct. Could you check what you get if you do `dir("/home/rstudio/external/hmdb_all_spectra/", full.names = TRUE, pattern = "ms_ms")` - maybe by defining the absolute path it works - I have no idea where the R workspace within docker is by default.

**Comment by gmhhope** [comment]
> Thanks Minghao!
> 
> hm, this is strange. I guess this has to do with the paths not being correct. Could you check what you get if you do `dir("/home/rstudio/external/hmdb_all_spectra/", full.names = TRUE, pattern = "ms_ms")` - maybe by defining the absolute path it works - I have no idea where the R workspace within docker is by default.

You are right. I guess there is something wrong with the path. It sometimes works, sometimes not (not because of the path I entered is wrong). I am not sure what are the issues but I will check myself. 

> ```
> #' Load the pre-generated Spectra object with spectra from HMDB.
> library(MsBackendHmdb)
> data("hmdb", package = "SpectraTutorials")
> ```

But your precompiled HMDB Rdata is great! Could I check what version you used to convert the HMDB? At least I know whether is enough to use your precompiled HMDB Rdata?

Thanks,
Minghao

**Comment by jorainer** [comment]
I downloaded the spectra data on 2021-01-01 from HMDB.
